proposals/0204-slashable-event-verification.md

```diff
@@ -0,0 +1,297 @@
+already been reported. Otherwise we create the account, with the slashing program
+as the owner. In this account we store the following:
+
+```rust
+struct ProofReport {
+  reporter: Pubkey,              // Fee payer, to allow the account to be closed
+  epoch: Epoch,                  // Epoch in which this report was created
+  pubkey: Pubkey,                // The pubkey of the node that committed the violation
+  slot: Slot,                    // Slot in which the violation occured
+  violation_type: u8,            // The violation type
+  proof: Vec<u8>                 // The serialized proof
+  proof_account: Option<Pubkey>, // Optional account where proof is stored instead
+}
+```
+
+The `DuplicateBlockProofData` is serialized into the `proof` field. This provides
+an on chain trail of the reporting process, since the `proof_account` supplied in
+the `DuplicateBlockProof` account could later be modified.
+
+The `pubkey` is populated with the `node_pubkey`. For future violation types that
+involve votes, this will instead be populated with the vote account's pubkey.
+The work in SIMD-0180 will allow the `node_pubkey` to be translated to a vote account
+if needed.
+
+Note that PDA's can only be created with a 10kb initial size.
+Although not a problem for `DuplicateBlockProofData`, if future proof types require
+more space, we allow the proof to be stored in a separate account, and linked back
+to the PDA using the `proof_account` field.
+
+---
+
+### Closing the incident report
+
+After the slashing violation has been processed by the runtime, the initial fee
+payer may wish to close their `ProofReport` account to reclaim the lamports.
+
+They can accomplish this via the `CloseProofReport` instruction which requires
+2 accounts:
+
+0. `report_account`: The PDA account storing the report: Writable, owned by the
+  slashing program
+1. `destination`: Writable account to reclaim the lamports
+
+`CloseProofReport` has an instruction data of 42 bytes, containing:
+
+- `0x01`, a fixed-value byte acting as the instruction discriminator
+- `violation_type`, a one byte value acting as the violation type discriminator
+- `slot`, an unaligned eight-byte little-endian unsigned integer indicating the
+  slot which was reported
+- `pubkey`, an unaligned 32 byte array representing the public key of the node
+  which was reported
+
+We abort if:
+
+- `violation_type` is not `0x00` (corresponds to `DuplicateBlock` violation)
+- Deriving the pda using `pubkey`, `slot`, and `ViolationType::DuplicateBlock`
+  as outlined above does not result in the adddress of `report_account`
+- `report_account` is not writeable
+- `report_account` does not deserialize cleanly to `ProofReport`
+- `report_account.reporter` is not a signer
+- `report_account.epoch + 3` is greater than the current epoch reported from
+  the `Clock` sysvar. We want to ensure that these accounts do not get closed before
+  they are observed by indexers and dashboards.
```

Comment:
Is there any other rationale for this number, like association to the leader schedule or time for stake / vote accounts to get penalized? It's ok if there isn't, but it would be good to spell it out.